engine/source/public/core/fileio.hpp

```diff
@@ -0,0 +1,47 @@
+﻿#pragma once
+#include <map>
+#include <string_view>
+#include <vector>
+
+namespace lumina
+{
+    class Engine;
+
+    class FileIO
+    {
+    public:
+        FileIO();
+        ~FileIO() = default;
+
+        FileIO(const FileIO&)            = delete;
+        FileIO(FileIO&&)                 = delete;
+        FileIO& operator=(const FileIO&) = delete;
+        FileIO& operator=(FileIO&&)      = delete;
+
+        enum class Directory
+        {
+            Assets,
+            Config,
+            Log
+        };
+
+        //Text file read/write
+        std::string ReadTextFile(Directory directory, const std::string& filePath);
+        bool WriteTextFile(Directory directory, const std::string& filePath, const std::string& data);
```

Comment:
this is a case where we do not need the attribute `[[nodiscard]]` since  we don't always care about the return value.